### PR TITLE
fix(developer): compiler crash when no project loaded

### LINKED ISSUE #9889: bug(developer): crash when compiling keyboard in 17.0
Sentry Issue: [KEYMAN-DEVELOPER-1DQ](https://keyman.sentry.io/issues/4560447942/?referrer=github_integration)

```
EAccessViolation: Exception EAccessViolation in module tike.exe at 00954998.
Access violation at address 00D54998 in module 'tike.exe'. Read of address 00000017.

  File "Keyman.Developer.System.KmcWrapper.pas", line 53, in Keyman.Developer.System.KmcWrapper.TKmcWrapper.Compile
    if ProjectFile.Project.Options.CompilerWarningsAsErrors then
  File "Keyman.Developer.System.Project.kmnProjectFileAction.pas", line 81, in Keyman.Developer.System.Project.kmnProjectFileAction.TkmnProjectFileAction.CompileKeyboard
    Result := w.Compile(Self, FileName, TargetFilename, Debug);
  File "Keyman.Developer.UI.Project.kmnProjectFileUI.pas", line 133, in Keyman.Developer.UI.Project.kmnProjectFileUI.TkmnProjectFileUI.CompileKeyboard
    Result :=
  File "Keyman.Developer.UI.Project.kmnProjectFileUI.pas", line 86, in Keyman.Developer.UI.Project.kmnProjectFileUI.TkmnProjectFileUI.DoAction
    pfaCompile:   Result := CompileKeyboard(FSilent);
  File "dmActionsKeyboardEditor.pas", line 457, in dmActionsKeyboardEditor.TmodActionsKeyboardEditor.actKeyboardCompileExecute
    if (ActiveProjectFile.UI as TProjectFileUI).DoAction(pfaCompile, False) and DebugReset then   // I4686
...
(30 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #9889.

This arises from a deprecated way of using Keyman Developer by starting Keyman Developer by opening a keyboard .kmn file from Explorer, which causes Developer to run in a 'projectless' mode. This was not tested in the new compiler integration, and that has caused this crash.

In 'projectless' mode, there is still a global 'untitled' project, but the files loaded do not 'belong' to it. The compiler settings are inherited from the global 'untitled' project in this case.

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/tike/compile/Keyman.Developer.System.KmcWrapper.pas [hunk 1]
```diff
@@ -50,11 +51,11 @@ function TKmcWrapper.Compile(
   cmdline := Format('"%s" build --log-format tsv --log-level info "%s"', [TKeymanDeveloperPaths.KmcPath, infile]);
   if outfile <> '' then
     cmdline := cmdline + Format(' --out-file "%s"', [outfile]);
-  if ProjectFile.Project.Options.CompilerWarningsAsErrors then
+  if FGlobalProject.Options.CompilerWarningsAsErrors then
```
Comment (darcywong00) [inline developer/src/tike/compile/Keyman.Developer.System.KmcWrapper.pas:54]: > In 'projectless' mode, there is still a global 'untitled' project, but the files loaded do not 'belong' to it. The compiler settings are inherited from the global 'untitled' project in this case.

Is `FGlobalProject` a superset of "has a .kpj" and `projectless` modes?

Comment (mcdurdin) [reply developer/src/tike/compile/Keyman.Developer.System.KmcWrapper.pas:54]: Yes, `FGlobalProject` is a global singleton, which always points to a valid, in memory project. That can be a project .kpj or the default untitled project for this fallback 'projectless' mode.

**Review by darcywong00 — APPROVED** [review]
lgtm
Perhaps a user test that the reported issue compiles :smile: 
https://community.software.sil.org/t/cant-compile-my-keyboard-with-tike-17/8070

**Comment by mcdurdin** [comment]
> Perhaps a user test that the reported issue compiles 😄
> https://community.software.sil.org/t/cant-compile-my-keyboard-with-tike-17/8070

Yeah I already tested locally and figured this one could be tested by the reporting users -- Bharani has enough work on right now!

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.204-alpha](https://keyman.com/downloads/releases/alpha/17.0.204)

**Pull request merged**